src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -39,22 +41,53 @@
 
 
 /** 
- * Servlet responsible for returning the iframe src for a user's primary calendar
+ * Servlet responsible for returning the link to the shared family calendar
 */
 public class CalendarServlet extends AbstractAppEngineAuthorizationCodeServlet {
-
-  static final String APP_NAME = "Household";
-
+  
   @Override
   protected void doGet(HttpServletRequest request, HttpServletResponse response)
       throws IOException {
-          
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Entity userInfoEntity = Utils.getCurrentUserEntity();
+
+    // If current user is not in a family, they cannot add a member
+    if (userInfoEntity == null) {
+        System.out.println("You do not belong to a family yet!");
+        response.setContentType("application/text");
+        response.getWriter().println("");
+        return;
+    }
+
+    Entity familyEntity = Utils.getCurrentFamilyEntity(userInfoEntity);
+
+    String calendarID = (String) familyEntity.getProperty("calendarID");
+
+    if (calendarID == null) {
+        response.setContentType("application/text");
+        response.getWriter().println("");
+        return;
+    }
+
     Calendar calendarService = Utils.loadCalendarClient();
 
-    CalendarListEntry primaryCalendarEntry = calendarService.calendarList().get("primary").execute();
+    CalendarListEntry calendarEntry;
+    try {
+        calendarEntry = calendarService.calendarList().get(calendarID).execute();
+    } catch (GoogleJsonResponseException e) {
+        // Create a new calendar list entry
```

Comment:
In what cases would we need to create a new calendar list entry?

Reply:
The calendar list contains all of the visible calendars. If we just share the new calendar with each member they cannot necessarily see it because it has not been added to their calendar list yet. So, if the shared calendar has not been added to a member's calendar list yet we create a new calendar list entry for them

Reply:
I see.